core/src/main/codegen/templates/Parser.jj

```diff
@@ -5050,6 +5050,86 @@ SqlIntervalQualifier TimeUnitOrName() : {
     }
 }
 
+  SqlIntervalQualifier TimeUnitOrNameForDateTrunc() : {
+    final TimeUnit timeUnit;
+    final SqlIntervalQualifier weekdayIntervalQualifier;
+    final SqlIdentifier unitName;
+}
+{
+    LOOKAHEAD(1)
+    timeUnit = TimeUnitForDateTrunc() {
+        return new SqlIntervalQualifier(timeUnit, null, getPos());
+    }
+|
+    weekdayIntervalQualifier = WeekWeekdaySqlIntervalQualifier() {
+        return weekdayIntervalQualifier;
+    }
+|
+    unitName = DateTruncIdentifiers() {
```

Comment:
I was following this description for the permitted operands: https://g3doc.corp.google.com/company/teams/googlesql/reference/functions-and-operators.md?cl=head#date-trunc
I was under the assumption that only these operands would be accepted and handled since I saw tests in `SqlOperatorTest` that fails when called with things like `microsecond`, `nanosecond`

Reply:
@olivrlee FYI that link is not available to people outside your VPN

Reply:
In CALCITE-5155 comment I wrote:

>  Previously, NANOSECOND and MILLISECOND were allowed in
> EXTRACT but no other functions. Now all functions that
>  accept time frames accept the same time frames (built-in
>  time intervals, identifiers for user-defined time frames,
>   and SQL_TSI_xxx which are defined in the JDBC standard but
>   are treated as identifiers until validation).

I think you should accept all time frames (including custom frames) in the parser. That should make the parser much simpler. We *could* reject certain frames at validate time, but I don't think we will. It's easier if we implement a superset of what BQ implements.

Reply:
Fixed link in comment above! thanks

Reply:
I will change parser to allow custom time frames and have the operands be a super set of what BigQuery enables